java/code/src/com/redhat/rhn/manager/system/SystemManager.java

```diff
@@ -2265,17 +2275,23 @@ public byte[] createProxyContainerConfig(User user, String proxyName, Integer pr
         config.put("ca_crt", rootCaCert);
 
         addTarEntry(tarOut, "config.yaml", YamlHelper.INSTANCE.dumpPlain(config).getBytes(), 0644);
-        /**
-         * config.yaml
-         */
+        // End of config.yaml
 
-        /**
-         * httpd.yaml
-         */
+        // httpd.yaml
         Map<String, Object> httpdRootConfig = new HashMap<>();
         Map<String, Object> httpdConfig = new HashMap<>();
 
-        Server proxySystem = getOrCreateProxySystem(user, proxyName, proxyPort);
+        Set<String> fqdns = new HashSet<>();
+        fqdns.add(proxyName);
+        if (proxyCertKey != null && proxyCertKey.getCertificate() != null) {
+            // Get the cnames from the certificate using openssl
+            SSLCertManager certManager = new SSLCertManager();
+            fqdns.addAll(certManager.getNamesFromSslCert(proxyCertKey.getCertificate()));
+        }
+        else {
+            fqdns.addAll(certData.getAllCnames());
+        }
```

Comment:
Nothing wrong with this snippet, but seems to me that javadoc for this part is not correct:
```
     * @param proxyCertKey proxy CRT and key pair
     * @param caPair the CA certificate and key used to sign the certificate to generate.
     *               Can be omitted if proxyCertKey is not provided
     * @param caPassword the CA private key password.
     *               Can be omitted if proxyCertKey is not provided
     * @param certData the data needed to generate the new proxy SSL certificate.
     *               Can be omitted if proxyCertKey is not provided
```

Shouldn't there everywhere be `Can be omitted if proxyCertKey IS provided`?